### DISCUSSION TITLE
Google maps is geolocating incorrectly in some countries

### DISCUSSION THREAD

**Discussion by GhaziTriki** [forkcms/forkcms#3347] [Ideas / Feature Requests]
## Type

- Non critical bug

## Problem description

ForkCMS puts the street number before the street name for geolocating addresses. The address number should come first.

## Steps to reproduce

Create a location with the following parameters:
- Street name: Rue de Annaba
- Street number: 16
- ZIP: 2074
- City: Mourouj

The geolocated address is incorrect.

## Expected behavior / Proposed solutions

Put the street number before the street name when geolocating.

**Comment by carakas** [comment]
can you provide a link to the documentation stating this?

**Comment by StijnVrolijk** [comment]
What exactly does Fork request?
[Streetname][Streetnumber][Zip][City] should be the logical structure but Google is pretty smart about this. I've used the geocoding api quite a few times and it handles everything i throw at it

**Comment by GhaziTriki** [comment]
I have found an element of answer at this link : http://stackoverflow.com/questions/7764244/correct-address-format-to-get-the-most-accurate-results-from-google-geocoding-ap

And in the official geocoder sample : http://googlemaps.github.io/js-v2-samples/geocoder/singlegeocode.html the address number comes first.

In all application I have developed the house number comes before the street name and I didn't see any coutnry that put the house number after the street name, is it the case in Belgium?

**Comment by carakas** [comment]
it is indeed :)

**Comment by carakas** [comment]
@GhaziTriki can you create a PR for this?

**Comment by GhaziTriki** [comment]
@carakas yes it should be an easy one (if I find the file where I need to look inside). Which is file is it?

**Comment by carakas** [comment]
it is in the engine of the location module, both back and frontend

**Comment by GhaziTriki** [comment]
@carakas is it also ok to put the street number before the street name in the backend form?

**Comment by StijnVrolijk** [comment]
@GhaziTriki I think it's best to leave it as is. When looking at other countries ( http://www.bitboost.com/ref/international-address-formats.html ) I think most of them use the street + number format.

**Comment by GhaziTriki** [comment]
I have created a pull request. However I think that the address creation should be more flexible. What we have now is that all fields are mandatory. The street number and the zip code should be optional as the street name and the city should be enough, or at least the street number.

**Comment by tijsverkoyen** [comment]
Personally, I think the address should be a textarea. People can enter the address in their own format.

**Comment by GhaziTriki** [comment]
@tijsverkoyen it is a more flexible solution that should work better

**Comment by carakas** [comment]
@GhaziTriki I closed your PR because the implementation would cause issues for people who have street + number in their country.

What I suggest is that you create a new pr and add a setting in the backend to the location module where you can set the format of the address. The default should be the format we have now, but then you can also add the format you need

**Comment by carakas** [comment]
the separate fields will be replaced with a single textarea for the address

**Comment by carakas** [comment]
take a look at https://github.com/commerceguys/addressing

**Comment by jeroendesloovere** [comment]
@GhaziTriki Are you willing to create a PR using https://github.com/commerceguys/addressing?

**Comment by GhaziTriki** [comment]
@jonasdekeukelaere I have some time. How do you think we can implement it?

**Comment by carakas** [comment]
@GhaziTriki we should implement it in a way that it works for everywhere, maybe just a text area? Or with that library, don't know exactly what is possible

**Comment by jonasdekeukelaere** [comment]
Hi @GhaziTriki - Did you start this? I can start this now otherwise, I'll add a textarea, as that way everyone is free to use whatever format they please.

**Comment by jeroendesloovere** [comment]
If I'm remembering correct we decided in the core meeting that a single textarea was NOT the way we should go.

**Comment by tijsverkoyen** [comment]
Euh, I remember that a texture was the way to go. As that allows people to enter/format the address as they want.

**Comment by jonasdekeukelaere** [comment]
So @jeroendesloovere I'll implement it as a textarea

**Comment by jeroendesloovere** [comment]
In the last core meeting (Tijs wasn't present then) we decided that if we use textarea. Lots of users wil lose the enormous features separate fields have to offer; "country", "zip", "city". 

I must say that we used those fields a lot to create smart functionality like you can see in the screenshot below:
<img width="1112" alt="schermafbeelding 2017-06-08 om 11 45 14" src="https://user-images.githubusercontent.com/588616/26922637-f4fd0ffc-4c3f-11e7-9d39-75ee2163302a.png">

**Comment by tijsverkoyen** [comment]
And how will you solve the issue of addresses in different address-formats?

**Comment by jeroendesloovere** [comment]
Looking at https://github.com/commerceguys/addressing,
they have the following fields:
<img width="744" alt="schermafbeelding 2017-06-08 om 11 46 49" src="https://user-images.githubusercontent.com/588616/26922713-2fe79d76-4c40-11e7-874c-b7a16be5d930.png">

So if I'm correct **merging "street + number/bus"** would be "address line 1" or "address line 2" (if there is an industrial zone to mention).

**Comment by tijsverkoyen** [comment]
The screenshots shows non-standard behaviour, and I don't see any Issues/Pull Requests that asks for extra functionality, so I don't see any reason to add extra dependencies, or make the interface more bloated.

**Comment by jeroendesloovere** [comment]
The actual problem of @GhaziTriki is that "streetnumber" should come before "street".
=> So why don't we create one textfield for "address line 1" (= street + number)
And leave the other fields untouched... (zip and city)

**Comment by tijsverkoyen** [comment]
In some countries (e.g.: UK) the zip isn't placed before the city.

**Comment by GhaziTriki** [comment]
As a reminder http://www.bitboost.com/ref/international-address-formats.html
Every country has its own address standard.
